PLATFORM 486 WIP

awalsh272 opened the pull request:
Adding functions for restCall and mainRecords into the appropriate repo, with some changes from the original made. Also added utility functions that are used in the data_loading script. Envisioning having separate py scripts for each survey form

awalsh272 committed:
feat: restCall and mainRecords into serverless

Adding functions for restCall and mainRecords into the appropriate repo, with some changes from the original made. Also added utility functions that are used in the data_loading script. Envisioning having separate py scripts for each survey form

awalsh272 committed:
feat: if/else logic in index, added envHealth

If else logic for choosing which function to run based on the specifier defined in event. Copied envHealth fcn to new py file, haven't adjusted it at all yet.

awalsh272 committed:
Delete secretz.py

awalsh272 committed:
feat gitignore

Hopefully I did this right

awalsh272 committed:
Merge branch 'PLATFORM-486-WIP' of https://github.com/puente-dr/serverless into PLATFORM-486-WIP

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/restCall.py:
```diff
@@ -0,0 +1,74 @@
+
+import secretz
+
+def lambda_handler(event, context):
```
This handler should replace the handler in hello-world>index.py

awalsh272 replied:
index.py is the lambda, with fcn name "handler". Other scripts are called/referenced in index

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/restCall.py:
```diff
@@ -0,0 +1,74 @@
+import requests
+import json
```
For all of your imports that are not native to python like request, pandas etc. You should create a requirement.txt that contains the name of every library import. Can show you what i mean if unclear. File should be located in hello-world folder.

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/utils.py:
```diff
@@ -0,0 +1,165 @@
+import pandas as pd
+import numpy as np
+import re
```
these two should also be added to requirements.txt

awalsh272 replied:
added a requirements.txt

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/secretz.py:
```diff
@@ -0,0 +1,3 @@
+APP_ID = "vBdTHqQU31IyLW5uYRDIWb8Ew8zCZGBzMqChugjr"
+
```
remove from PR

awalsh272 replied:
I think I did this with a new commit?

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/restCall.py:
```diff
@@ -0,0 +1,74 @@
+    #    This function makes a REST Get call to Back4App, returning a pandas df"""
+
+    url="https://parseapi.back4app.com/classes/"
+    specifier = "SurveyData"
```
should try to read the specficer from the event. The request coming in should look like this:

{
   surveyingOrganization: "Puente",
   parseClass: "SurveyData"
}

So you would want to insert those parameters in for specifier and where "OWS" is. Let me know if that needs clearing up.

awalsh272 replied:
Should be good now

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/restCall.py:
```diff
@@ -0,0 +1,74 @@
+      "order": "-updatedAt", "limit": 200000, "where":{
+          json.dumps({
+            "surveyingOrganization": {
+                "$in": ['OWS']
```
surveyingOrganization request param instead of OWS

awalsh272 replied:
Should be good now

joseph-mccombs reviewed (changes requested):
A few little things. god work!

awalsh272 committed:
feat: lots

Added evalmedical, some changes to the writing process. Better return responses, reading in from event

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/index.py:
```diff
@@ -1,9 +1,18 @@
+  survey_org = event["surveyingOrganization"]
+  specifier = event["specifier"]
+
+  if specifier == "SurveyData":
```
Right idea here. For some of the checks we're going to be doing, I want to make the rest call standard prior to doing all of the work for each form like cleaning. I essentially want:

restCall(specifier, survey_org)

* we will eventually do some checks here but not currently

Then break it out into the other form specfic work.

joseph-mccombs commented on puente-data-exporter/lambdas/hello-world/index.py:
```diff
@@ -1,9 +1,18 @@
+  elif specifier == "EvaluationMedical":
+    response = evalMedical(survey_org)
+  else:
+    print("Pick a valid specifier")
```
We do have vitals and customForms too :)

awalsh272 committed:
feat: requirements.txt

Used pipreqs to generate a requirements file

awalsh272 committed:
feat: restCall outside of cleaning fcns

awalsh272 committed:
feat: saving cleaned mainRecords locally

Running the function in command line, set the appropriate parameters in event.json to be read in. Had to do some json stuff to read it in properly. Then saving to a local file along a specified path of ./{survey_org}

joseph-mccombs committed:
feat: reorganize directory, create cloudformation with virtualenv

hopetambala committed:
Merge pull request #6 from jamccomb92/PLATFORM-486-WIP

feat: reorganize directory, create cloudformation with virtualenv

awalsh272 committed:
feat: result to csv on s3

self explanatory, but I'm getting an error that events/event.json doesn't exist when I run this using sam local invoke

awalsh272 committed:
Merge branch 'PLATFORM-486-WIP' of https://github.com/puente-dr/serverless into PLATFORM-486-WIP

awalsh272 committed:
trying to get it to run locally

from inside puente-data-exporter I'm running the command:

sam
local invoke LambdaFunction -e ./lambdas/data-exporter/events/event.json -t templates/cloudformation.yaml

and getting the error

[ERROR] Runtime.ImportModuleError: Unable to import module 'lambdas/data-exporter/indeTraceback (most recent call last):exporter/index'

after some stuff about decompressing the zip, using a local image and mounting something

hopetambala committed:
chore: deploy

hopetambala committed:
Update README.md

hopetambala committed:
chore: move to workflows

hopetambala committed:
Update README.md

hopetambala committed:
Update deploy.sh

hopetambala committed:
Update deploy.sh

hopetambala committed:
Update deploy.sh

hopetambala committed:
Create cloudformation-dev.yaml

hopetambala committed:
Update deploy.sh

hopetambala committed:
Update deploy.sh

hopetambala committed:
Merge branch 'PLATFORM-486-WIP' into dev

hopetambala committed:
Update cloudformation-dev.yaml

hopetambala committed:
chore: fix deploy script

hopetambala committed:
Update cloudformation-master.yaml

hopetambala committed:
Merge pull request #7 from puente-dr/dev

Dev
